src/amo/pages/Feedback/index.js

```diff
@@ -0,0 +1,478 @@
+  abuseReport: AddonAbuseState | null,
+|};
+
+type DefaultProps = {|
+  _window: typeof window | Object,
+  _config: typeof config,
+|};
+
+type InternalProps = {|
+  ...PropsFromState,
+  ...DefaultProps,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  // `match` is used in `mapStateToProps()`
+  // eslint-disable-next-line react/no-unused-prop-types
+  match: {|
+    ...ReactRouterMatchType,
+    params: {| addonIdentifier: string |},
+  |},
+|};
+
+type FormValues = {|
+  name: string,
+  email: string,
+  text: string,
+  category: string | null,
+|};
+
+type State = {|
+  ...FormValues,
+  successMessage: string | null,
+|};
+
+type Reason = {|
+  value: string,
+  label: string,
+  help: string,
+|};
+
+export const getCategories = (
+  i18n: I18nType,
+): {
+  report: Array<Reason>,
+  feedback: Array<Reason>,
+} => {
+  return {
+    feedback: [
+      {
+        value: 'does_not_work',
+        label: i18n.gettext(
+          'It doesn’t work, breaks websites, or slows down Firefox',
+        ),
+        help: i18n.gettext(
+          "Example: Features are slow, hard to use, or don’t work; parts of websites won't load or look unusual",
+        ),
+      },
+      {
+        value: 'not_wanted',
+        label: i18n.gettext('it’s SPAM'),
+        help: i18n.gettext(
+          'Example: An application installed it without my permission',
+        ),
+      },
+    ],
+    report: [
+      {
+        value: 'policy_violation',
+        label: i18n.gettext('It violates Add-on Policies'),
+        help: i18n.gettext(
+          'Example: I didn’t want it or It compromised my data without informing or asking me, or it changed my search engine or home page without informing or asking me.',
+        ),
+      },
+      {
+        value: 'hate_speech',
+        label: i18n.gettext(
+          'It contains hateful, violent, deceptive, or other inappropriate content',
+        ),
+        help: i18n.gettext('Example: contains racist imagery'),
+      },
+      {
+        value: 'illegal',
+        label: i18n.gettext(
+          'It violates the law or contains content that violates the law',
+        ),
+        help: i18n.gettext(
+          'Example: Copyright or Trademark Infringement, Fraud',
+        ),
+      },
+      {
+        value: 'other',
+        label: i18n.gettext('Something else'),
+        help: i18n.gettext(
+          'Anything that does’t fit into the other categories',
+        ),
+      },
+    ],
+  };
+};
+
+export class FeedbackBase extends React.Component<InternalProps, State> {
+  static defaultProps: DefaultProps = {
+    _window: typeof window !== 'undefined' ? window : {},
+    _config: config,
+  };
+
+  constructor(props: InternalProps) {
+    super(props);
+
+    const { dispatch, errorHandler, currentUser } = props;
+
+    this.state = {
+      successMessage: null,
+      ...this.getFormValues(currentUser),
+    };
+
+    dispatch(setViewContext(VIEW_CONTEXT_HOME));
```

Comment:
why is this needed?

Reply:
it may not be - it was just in a many other components so I assumed it was important 🤷